### PR TITLE
Feature add forms

### PR DESCRIPTION
**Pull request by Graciano1997**
In this Milestone I did the following activities:

Created a method called current_user in  ApplicationController that returns the first user from the database.

Created forms to perform the following functions:
- Created a Post on behalf of the current_user.
- Created a comment on behalf of the current_user
- Allow Users to add likes to Posts.

Update readme
Fix linter issues

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+
+  def create
+    @post = Post.create(author: current_user, title: params[:post][:title], text: params[:post][:text],
+                        comments_counter: 0, likes_counter: 0)
+    if @post.new_record?
+      redirect_to '/users/1/posts/new', flash: { wrong: 'Upps! Post was not created.' }
+    else
+      redirect_to '/users/1/posts/new', flash: { success: 'Post was successfully created.' }
+    end
   end
```
Comment (Abenezer-Tilahun) [inline app/controllers/posts_controller.rb:27]: - [ Optional ] Kindly I recommend you to use `post_params ` as **private method** .This method is used to handle strong parameters in Rails, which is a security feature that prevents assigning request parameters to objects unless they have been explicitly permitted. For more information you can refer this [Link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) Just like the below sample 

```
private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    @comment = Comment.create(text: params[:comment][:text], user: current_user, post_id: params[:post_id])
+    if @comment.new_record?
+      redirect_to "/users/#{params[:user_id]}/posts/#{params[:comment][:url_id]}",
+                  flash: { wrong: 'Upps! Comment was not created.' }
+    else
+      redirect_to "/users/#{params[:user_id]}/posts/#{params[:comment][:url_id]}",
+                  flash: { success: 'Comment was successfully created.' }
+    end
+  end
+end
```
Comment (Abenezer-Tilahun) [inline app/controllers/comments_controller.rb:18]: - [ Optional ] Kindly I recommend you to use `comment_params ` as **private method** .This method is used to handle strong parameters in Rails, which is a security feature that prevents assigning request parameters to objects unless they have been explicitly permitted. For more information you can refer this [Link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) Just like the below sample 

```
private 
   def comment_params
    params.require(:comment).permit(:text)
  end

```

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Graciano1997 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages.
✔️ Correctly make current user data available to all controllers.
✔️ Correctly Create forms to Creates a Post and comment on behalf of the current_user.
✔️ Correctly Allow Users to add likes to Posts.
✔️ Good job for Implemented Blog app - Add forms.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Comment by Graciano1997** [comment]
@Abenezer-Tilahun  thank you for recognizing all my efforts!  🥂  🥂

**Pull request merged**